src/dbnode/storage/cluster/database.go

```diff
@@ -151,6 +151,46 @@ func (d *clusterDB) Close() error {
 	return d.stopActiveTopologyWatch()
 }
 
+func (d *clusterDB) IsBootstrappedAndDurable() bool {
+	if !d.Database.IsBootstrapped() {
+		return false
+	}
+
+	_, ok := d.topo.(topology.DynamicTopology)
+	if !ok {
+		// If the topology is not dynamic, then the only thing we care
+		// about is whether the node is bootstrapped or not because the
+		// concept of durability as it relates to shard state doesn't
+		// make sense if we're using a static topology.
```

Comment:
Hm is this true?  Should we perhaps be calling `d.Database.IsBootstrappedAndDurable()` so that at least in the case where the topology is not dynamic it's had at least one snapshot, etc?

Reply:
spoke offline, existing implementation should be good but I added more context in the comments.